### DISCUSSION TITLE
Maybe we can add a date-range to the beatmap listing in client?

### DISCUSSION THREAD

**Discussion by Inconsist** [ppy/osu#22308] [Ideas]
I recently wanted to download a map and all I knew is that it was roughly ranked around 03.2021 with the beatmap listing there seems to be no way to do date selection by typing in filters, so scrolling I went. A few minutes and 5gb of ram now occupied with over a year of beatmaps later, I arrived in may, fps where down to 20-50 form 1000 I steadily have in gameplay. Frame time spikes all over the place.

I am by absolutely no means a person qualified to judge this but the option to set a daterange must also reduce load for the servers since they only have to send out presorted packages of beatmaps to me, instead of EVERYTHING up to the point im interested in.

So my suggested fix to this would be to eliminate the need for the servers to send me all the beatmaps I didnt need in the first place, by letting me set a date range. Maybe a similar thing could be possible for all other paramters too, but dates would offer the greatest improvement in useability in my eyes and if sorting for star ranges in this stage would put more load on the servers than can be saved in bandwith then ditch that. 

Idea for UI implementation in alignment with the new design as follows:

![Screenshot 2023-01-20 213532](https://user-images.githubusercontent.com/87947227/213818715-4d8d5dfb-72ce-47e6-a3cc-1fc2c2a14ec5.png)

I know the possibility to sort by dates up/down is already there but that really only allows me to view a couple thousand beatmaps deep from either starting side. People with less ram probably even less. 

And due to the fact that that menue is now not populated with so many maps the performance issues that occur this deep into the listings might also be eliminated too. 

Unrelated but similar idea to improve beatmap listing in #22309

**Comment by bdach** [comment]
1. You can use a text filter for rank date (`ranked>2023-03-01`) for now.
2. https://github.com/ppy/osu/discussions/17339 for UI.
3. Beatmap listing performance needs work, it's a known issue. Should improve with pooling.

**Reply by Inconsist** [reply]
Is there a wiki for these filters? Never seen this one and other filters like artist=" " didnt work for me if I didnt mess it up, so i thought you cant filter in listing.

**Reply by Inconsist** [reply]
Is there a thread about pooling? What is it and does it solve tha ram issues too, or just the fps?
